tensorflow/python/data/experimental/kernel_tests/data_service_ops_test.py

```diff
@@ -648,6 +648,86 @@ def testRegisterDifferentDatasets(self):
+    num_sizes = 10
+    size_repeats = 5
+    numbers = [1 * i for i in range(num_sizes)] * size_repeats
+    ds = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds = self.make_distributed_dataset(
+        ds, cluster_1, processing_mode="parallel_epochs")
+    ds = ds.map(lambda x: x + 1)
+    ds = self.make_distributed_dataset(
+        ds, cluster_2, processing_mode="distributed_epoch")
+
+    error_regex = "Cannot create a split provider for dataset of type DataServiceDataset"  # pylint: disable=line-too-long
+    with self.assertRaisesRegex(errors.UnimplementedError, error_regex):
+      self.getDatasetOutput(ds)
+
+  @combinations.generate(test_base.eager_only_combinations())
+  def testDistributeDistributedToParallel(self):
+
+    cluster_1 = self.create_cluster(num_workers=1)
+    cluster_2 = self.create_cluster(num_workers=1)
+    num_sizes = 10
+    size_repeats = 5
+    numbers = [1 * i for i in range(num_sizes)] * size_repeats
+    ds = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds = self.make_distributed_dataset(
+        ds, cluster_1, processing_mode="distributed_epoch")
+    ds = ds.map(lambda x: x + 1)
+    ds = self.make_distributed_dataset(
+        ds, cluster_2, processing_mode="parallel_epochs")
+
+    self.assertDatasetProduces(
+        ds, [i + 1 for i in numbers], assert_items_equal=True)
+
+  def testParallelZippedDistributedDatasets(self):
+
+    cluster_1 = self.create_cluster(num_workers=1)
+    cluster_2 = self.create_cluster(num_workers=1)
+    cluster_3 = self.create_cluster(num_workers=1)
+    num_sizes = 10
+    size_repeats = 5
+    numbers = [1 * i for i in range(num_sizes)] * size_repeats
+    ds1 = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds1 = self.make_distributed_dataset(ds1, cluster_1)
+
+    ds2 = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds2 = self.make_distributed_dataset(ds2, cluster_2)
+
+    ds3 = dataset_ops.Dataset.zip((ds1, ds2))
+    ds3 = self.make_distributed_dataset(
+        ds3, cluster_3, processing_mode="parallel_epochs")
+
+    self.assertDatasetProduces(
+        ds3, list(zip(numbers, numbers)), assert_items_equal=True)
+
+  def testDistributedZippedDistributedDatasets(self):
+
+    cluster_1 = self.create_cluster(num_workers=1)
+    cluster_2 = self.create_cluster(num_workers=1)
+    cluster_3 = self.create_cluster(num_workers=1)
+    num_sizes = 10
+    size_repeats = 5
+    numbers = [1 * i for i in range(num_sizes)] * size_repeats
+    ds1 = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds1 = self.make_distributed_dataset(ds1, cluster_1)
+
+    ds2 = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds2 = self.make_distributed_dataset(ds2, cluster_2)
+
+    ds3 = dataset_ops.Dataset.zip((ds1, ds2))
+    ds3 = self.make_distributed_dataset(
+        ds3, cluster_3, processing_mode="distributed_epoch")
+
+    error_regex = "Cannot create a split provider for dataset of type ZipDataset"  # pylint: disable=line-too-long
```

Comment:
@aaudiber  I was only looking at the tests in `tensorflow/python/data/experimental/kernel_tests/data_service_ops_test` and didn't find any test which used `distributed_epoch` and raised the `split provider` type error on a ZipDataset. Thus, I thought of adding one. Please correct me if I am wrong.

Reply:
Ah you're right. The test as it stands is a little misleading though, since it makes it look like the error happens specifically because the datasets inside the zipped dataset are distributed. But in fact any zip-based dataset can't be distributed, so we could simplify the test to use simple datasets (e.g. `dataset_ops.Dataset.range(10)`) inside the zip.

Reply:
addressed in the commit.